### PR TITLE
(WIP) Add deployment on tag

### PR DESCRIPTION
**Pull request by michikrug**
This PR adds a GitHub action to deploy the code automatically to Google Cloud Function when a tag was pushed.

Currently, it refers to a test function, we created and needs to be changed to the production one, ones this was evaluated.

Additionally, I added release-drafter to have nice drafts within GitHub for production releases.

### CODE REVIEW

**Comment by Legion2** [comment]
When the CI jobs are defined in different files it's easier to define the individual triggers for the workflows. So you don't need the conditions for the jobs.

**Comment by michikrug** [comment]
> When the CI jobs are defined in different files it's easier to define the individual triggers for the workflows. So you don't need the conditions for the jobs.

But can I still require the other ones to succeed then?

**Comment by Legion2** [comment]
> Bcs. I only want to deploy, if tests and analysis was successful...

When you can only make releases from main branch and you can only merge into main branch if the test and analysis is passing, then you don't have to run the tests again before releasing it.

**Comment by michikrug** [comment]
But a person with write access could also just push into the main branch and accidentally break stuff.

If we do not consider this "human error", it would ofc. work as you described it.

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: javascript
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
+
+  deployment:
+    if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'release'
```
Comment (Legion2) [inline .github/workflows/ci-cd.yml:65]: does this deploy on tags and releases? so there are two release when a tagged release is created?

Comment (michikrug) [reply .github/workflows/ci-cd.yml:65]: The idea I currently had in mind was to have a "stage" version that is deployed when a tag is set and the "production" one that is only deployed on an explicit release.
So far not really sure, to be honest, if this makes sense or works at all.

**Pull request merged**